src/app/SlaveInterface/SlaveInterface.c

```diff
@@ -0,0 +1,88 @@
+#include "HAL_SlaveChips.h"
+
+#include "FaultManager.h"
+#include "SlaveInterface.h"
+
+/**
+ * Get the latest voltage and draining information from the slave chips.
+ * Set faults if communication errors occur.
+ * battery_model [out] - destination for updated cell info
+ */
+void SlaveInterface_read_cell_info(BatteryModel_t* battery_model)
+{
+    float voltages[NUM_SERIES_CELLS];
```

Comment:
Something worth considering in the future could be not using floats for cell voltages since I'm pretty sure we don't have floating-point hardware. So any floating-point arithmetic is being done in software which can be pretty slow.

We could basically do fixed decimal encoding. So 345 would represent 3.45V. I doubt we'll have problems since we are running a pretty fast clock, but something to keep in mind if we face problems

Reply:
Yeah we'll see how it goes and use a library like iqmath if we need to